SmallCommitsWorkshopTests/Services/FizzBuzzServiceTests.cs

```diff
@@ -46,5 +46,29 @@ internal sealed class FizzBuzzServiceTests {
 				actual: m_fizzBuzzService.Calculate( number: number )
 			);
 		}
+
+		[TestCase( 1, "some-fizz", "some-buzz", ExpectedResult = "1" )]
+		[TestCase( 3, "some-fizz", "some-buzz", ExpectedResult = "some-fizz" )]
+		[TestCase( 3, "", "some-buzz", ExpectedResult = "" )]
+		[TestCase( 3, null, "some-buzz", ExpectedResult = "" )]
+		[TestCase( 5, "some-fizz", "some-buzz", ExpectedResult = "some-buzz" )]
+		[TestCase( 5, "some-fizz", "", ExpectedResult = "" )]
+		[TestCase( 5, "some-fizz", null, ExpectedResult = "" )]
+		[TestCase( 15, "some-fizz", "some-buzz", ExpectedResult = "some-fizzsome-buzz" )]
+		[TestCase( 15, "", "some-buzz", ExpectedResult = "some-buzz" )]
+		[TestCase( 15, null, "some-buzz", ExpectedResult = "some-buzz" )]
+		[TestCase( 15, "some-fizz", "", ExpectedResult = "some-fizz" )]
+		[TestCase( 15, "some-fizz", null, ExpectedResult = "some-fizz" )]
+		[TestCase( 15, "", "", ExpectedResult = "" )]
+		[TestCase( 15, "", null, ExpectedResult = "" )]
+		[TestCase( 15, null, "", ExpectedResult = "" )]
+		[TestCase( 15, null, null, ExpectedResult = "" )]
+		public string Calculate__int__string__string( int number, string fizz, string buzz ) {

```

Comment:
Curious: is `__int__string__string` a NUnit thing? Or a convention you usually use?

Reply:
It's a convention I've adopted for naming test methods when there are overloads.  The style itself comes from how JNI methods we resolved by RIM's JVM: a java method like `private native void DoStuff( string foo, int bar )` would be implemented in C++ as something like `void ClassName__DoStuff__string__int`.

Somewhere along the way I got it in my head that this was also the style used by [C++ name mangling](https://en.wikipedia.org/wiki/Name_mangling#C++), but a little research proves that wrong 😳.  Having said that, I do _use_ this style for the same reason that C++ name mangling exists :grin:.